components/director/internal/graphql/scalars/annotations_test.go

```diff
@@ -0,0 +1,59 @@
+package scalars
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAnnotations_UnmarshalGQL(t *testing.T) {
+	//given
+	a := Annotations{}
+	fixAnnotations := map[string]interface{}{
+		"annotation1": "val1",
+	}
+	expectedAnnotations := Annotations{
+		"annotation1": "val1",
+	}
+
+	//when
+	err := a.UnmarshalGQL(fixAnnotations)
+
+	//then
+	require.NoError(t, err)
+	assert.Equal(t, a, expectedAnnotations)
+}
+
+func TestAnnotations_MarshalGQL_Int(t *testing.T) {
```

Comment:
As those two following test functions are similar in the implementation only differs by given arguments you can reuse the implementation by providing the test payload and expectations as a list and iterating over those values.

For example https://tutorialedge.net/golang/improving-your-tests-with-testify-go/#combining-testify-with-table-driven-tests

Reply:
done